src/files/BrsFile.ts

```diff
@@ -901,11 +901,15 @@ export class BrsFile {
                 // the next symbol to check will be the return value of this function
                 symbolType = getTypeFromContext(symbolType.returnType, typeContext);
                 if (tokenFoundCount < tokenChain.length) {
-                    // We're still
+                    // We still have more tokens, but remember the last known reference
                     symbolTypeBeforeReference = symbolType;
                 }
             }
 
+            if (isArrayType(symbolType) && tokenUsage === TokenUsage.ArrayReference) {
```

Comment:
If this token was using an array reference (e.g `token[i]`)  then the actual type it represents is the type of the elements in the array